Sources/StreamChatUI/Utils/ImageCDN.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+public enum ImageCrop: String {
+    case top, bottom, left, right, center
+}
+
+public enum ImageResize: String {
+    case crop, scale, fill
+}
+
+/// Caching key provider is providing customised (filtered) key to be used for caching.
+///
+/// The URL can contain extra parameters that are changing every time. Those parameters are effectively blocking the caching.
+/// By providing customised (filtered) key, the cache will work as expected.
+public protocol ImageCDN {
```

Comment:
This works more like a URL provider than CDN, so I suggest renaming this to `ImageURLProvider` or something like it, basically removing `CDN` from the name

Reply:
The problem is that it's not providing URL, is providing URL modifiers/processors, so I was struggling with the name a little. I used the one suggested by @VojtaStavik in the task

Reply:
For sure `ImageCDN` is not the best name but I think it kinda works 🤷‍♂️ You'd need to implement these methods if you use your own CDN for images. We'll probably add more methods here as we add new features.